### ISSUE TITLE
Freeze / crash occurring when batch size is reduced

### ISSUE THREAD

**Issue by drjubbs** [tensorflow/tensorflow#44459]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): **Yes**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): **Windows 10 Home 1909**
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): **binary**
- TensorFlow version (use command below): **2.3.1**
- Python version: **3.7.5**
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: **10.1 / cudnn-10.1-windows10-x64-v7.6.5.32**
- GPU model and memory: **GeForce GTX 1660 Ti computeCapability: 7.5, coreClock: 1.455GHz coreCount: 24 deviceMemorySize: 6.00GiB deviceMemoryBandwidth: 268.26GiB/s**

**Describe the current behavior**
When running with large batch sizes in `model.fit()`, code completes successfully. With smaller batch sizes, the training crashes/freezes in the middle of an epoch.

**Describe the expected behavior**
Smaller batch sizes do not cause TF to hang or crash.

**Standalone code to reproduce the issue**

```
# -*- coding: utf-8 -*-
"""Reproduce crash during fit"""
import numpy as np
from tensorflow.keras.layers import Dense
from tensorflow.keras.models import Sequential


model = Sequential()
model.add(Dense(9, activation='relu', input_dim=125))
model.add(Dense(31, activation='softmax'))
model.compile(loss='categorical_crossentropy',
              optimizer='adam',
               metrics=['accuracy'])

X_train=np.random.rand(3225, 125)
y_train=np.random.rand(3225, 31)

# This works
model.fit(X_train[:,:], y_train[:,:], 
          epochs=100, 
          batch_size=X_train.shape[0], 
          verbose=1)

# This crashes
model.fit(X_train[:,:], y_train[:,:], 
          epochs=100, 
          batch_size=100, 
          verbose=1)
```

The exact point it freezes seems to vary based on the random seed, here is an example. The shell (`cmd.exe`) is frozen and must be killed through task manager.

```
Epoch 11/100
33/33 [==============================] - 1s 18ms/step - loss: 607.7234 - accuracy: 0.0332
Epoch 12/100
24/33 [====================>.........] - ETA: 0s - loss: 645.0050 - accuracy: 0.0288
```

**Comment by Saduf2019** [comment]
@drjubbs 
I ran the code shared and do not see the issue reported on tf 2.3 or nightly, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/fc06f34a494a4c162fc3a9527cb53f63/untitled457.ipynb),[ nightly](https://colab.research.google.com/gist/Saduf2019/0f09432a9a5f2a5c4826a86d388253bd/untitled458.ipynb)

**Comment by drjubbs** [comment]
FYI - I was using Nvidia "Game Ready" drivers (441.22), a newer one (457.09) was released, I upgraded, and I no longer get the crash. Not sure if you test against the "Game Ready" versions but this may have been the cause?

**Comment by Saduf2019** [comment]
@drjubbs 
As you as see this is not a bug because of tensorflow and i was able to run your code without any crashing, as the issue is resolved please move this closed status.

**Comment by Saduf2019** [comment]
Moving to closed status as resolved.
